test/test_optim.py

```diff
@@ -1057,7 +1057,11 @@ def fwd_bwd(optim, w, b, i):
 
             # Prime the optimizer
             for _ in range(10):
-                optimizer.step(closure)
+                if torch._dynamo.is_compiling() and not optim_info.step_requires_closure:
```

Comment:
```suggestion
                if not optim_info.step_requires_closure:
```
let's keep the testing between PYTORCH_TEST_WITH_DYNAMO vs not as close as possible

Reply:
So in this case, you *want* to run the disabled optimizer? This will be the eager optimizer again.

Reply:
The idea behind the PYTORCH_TEST_WITH_DYNAMO tests seem to mean "hey adding dynamo to our tests should still pass". If that's the spirit, then we should not special case the test on that flag being enabled.

Reply:
This is the closure thing, the tests won't mean anything if we run with closure, because dynamo will fall back. We won't actually be testing anything related to compiled optimizer

Reply:
Testing that dynamo falls back correctly is valid, and that it traces correctly is also valid.

And to be clear, my suggestion allows the closure separation for eager tests but doesn't touch what's currently getting tested in dynamo, so the dynamo part runs everything with the separated closure call--which I believe is also what you want.

Reply:
Oh oops, sounds good!